### PR TITLE
Extract user metrics from cache

### PR DESCRIPTION
**Pull request by AlecErasmus**
Extract user metrics from source cred cache

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -8,5 +8,6 @@ NEXTAUTH_SECRET=RAMDON_GENERATED_BASE64
 
 CRED_MANAGER_ROUTE="http://localhost:8080"
 SOURCECRED_INSTANCE_PATH="/instance"
+SOURCECRED_CACHE_DB_PATH="/instance/cache/sourcecred/github"
```
Comment (sudoFerraz) [inline .env.example:11]: I guess this also needs to be added on the secrets then right? Double check bc I think it also needs to be added on the action itself / .ENV file

File: cred-manager/package.json [hunk 2]
```diff
@@ -9,6 +9,8 @@
     "prebuild": "rimraf dist",
     "build": "nest build",
     "format": "prettier --write \"src/**/*.ts\" \"test/**/*.ts\"",
+    "sqllite:generate": "npx prisma generate --schema prisma-lite/schema.prisma",
+    "sqllite:push": "npx prisma db push --schema prisma-lite/schema.prisma",
```
Comment (sudoFerraz) [inline cred-manager/package.json]: From what I understood, we are not using the prisma driver with the sqlite db right? And we are using the better-sqlite for that, if that's the case, we can delete these scripts

File: cred-manager/prisma-lite/schema.prisma [hunk 3]
```diff
@@ -0,0 +1,139 @@
+generator client {
+  provider = "prisma-client-js"
+  output   = "../node_modules/@prisma-lite/client"
+}
+
+datasource db {
+  provider = "sqlite"
+  url      = "file:/Users/alecerasmus/development/web3/sourcecred/armitage-dashboard/cred-manager/cache/github_armitage-labs_armitage-monorepo.db"
```
Comment (sudoFerraz) [inline cred-manager/prisma-lite/schema.prisma]: This URL is from your local environment.

Will we need to parametrize this?

File: cred-manager/src/sourcecred/service/cache.service.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import { readdirSync } from 'fs';
+
+@Injectable()
+export class CacheService {
+  private readonly logger = new Logger(CacheService.name);
+
+  public getSqlLiteDatabase(dbPath: string): any[] {
+    const dbFiles = this.findFilesWithExtension(dbPath, '.db');
+    return dbFiles.map((db) => {
+      // return new Database(db, { verbose: console.log });
```
Comment (sudoFerraz) [inline cred-manager/src/sourcecred/service/cache.service.ts]: comment

File: cred-manager/src/sourcecred/service/sourcecred.service.ts [hunk 5]
```diff
@@ -62,13 +69,25 @@ export class SourceCredService {
...
+      this.sourceCredCacheDbPath,
+    );
+
     await this.deleteContribution(teamId);
     const contribution = await this.saveContribution(teamId, credGrainView);
     await this.saveUserScore(contribution.id, userCredDtoArray);
+
+    for (let i = 0; i < databases.length; i++) {
+      const database = databases[i];
+      const credMetrics = await this.loadLocalScMetrics(database);
```
Comment (sudoFerraz) [inline cred-manager/src/sourcecred/service/sourcecred.service.ts:94]: We should start breaking down this "calculateCredScores" function, it's getting too big now

File: cred-manager/src/sourcecred/service/sourcecred.service.ts [hunk 6]
```diff
@@ -100,6 +119,7 @@ export class SourceCredService {
     let success = true;
     const cmdList = [
       `export SOURCECRED_GITHUB_TOKEN=${gitHubToken} && cd ${this.sourceCredPath} && yarn sourcecred go`,
+      `ls -la ${this.sourceCredPath}/cache/sourcecred/github`,
```
Comment (sudoFerraz) [inline cred-manager/src/sourcecred/service/sourcecred.service.ts]: leftover from debugging?

**Review by sudoFerraz — CHANGES_REQUESTED** [review]

File: .env.example (hunk 1 above)
Comment (AlecErasmus) [reply .env.example:11]: yes

**Review by sudoFerraz — APPROVED** [review]

**Pull request merged**